nifi-commons/nifi-expression-language/src/main/antlr3/org/apache/nifi/attribute/expression/language/antlr/AttributeExpressionParser.g

```diff
@@ -77,8 +77,9 @@ zeroArgString : (TO_UPPER | TO_LOWER | TRIM | TO_STRING | URL_ENCODE | URL_DECOD
 oneArgString : ((SUBSTRING_BEFORE | SUBSTRING_BEFORE_LAST | SUBSTRING_AFTER | SUBSTRING_AFTER_LAST | REPLACE_NULL | REPLACE_EMPTY |
 				PREPEND | APPEND | STARTS_WITH | ENDS_WITH | CONTAINS | JOIN | JSON_PATH | FROM_RADIX) LPAREN! anyArg RPAREN!) |
 			   (TO_RADIX LPAREN! anyArg (COMMA! anyArg)? RPAREN!);
-twoArgString : ((REPLACE | REPLACE_FIRST | REPLACE_ALL | IF_ELSE) LPAREN! anyArg COMMA! anyArg RPAREN!) |
-			   ((SUBSTRING | FORMAT) LPAREN! anyArg (COMMA! anyArg)? RPAREN!);
+twoArgString : ((REPLACE | REPLACE_FIRST | REPLACE_ALL | IF_ELSE | ) LPAREN! anyArg COMMA! anyArg RPAREN!) |
+			   ((SUBSTRING | FORMAT) LPAREN! anyArg (COMMA! anyArg)? RPAREN!) | (PAD_LEFT LPAREN! anyArg (COMMA! anyArg)? RPAREN!) |
```

Comment:
can't we just add `PAD_LEFT` and `PAD_RIGHT` to the function names here?

Reply:
My intent here is to allow both 1 or 2 parameters, if I just add them to the declared functions (i.e. in the list, after the IF_ELSE function), won't I allow only two parameters?

Reply:
yes, but if you put them in the list with `SUBSTRING | FORMAT`, you allow 2 parameters, don't you?

Reply:
Yes, maybe I am a bit confused about how Antlr3 works, I tried this
```
((SUBSTRING | FORMAT | PAD_LEFT | PAD_RIGHT) LPAREN! anyArg (COMMA! anyArg)? RPAREN!));
```
But it seems I am not able to build the parser

Reply:
mmmh...what do you mean? moreover in the line above you  left a useless `|` I think

Reply:
Ok I managed to make it after all